src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -269,14 +268,44 @@ def __read_kv_from_app_service(cmd, appservice_account, prefix_to_add=""):
             cmd, resource_group_name=appservice_account["resource_group"], name=appservice_account["name"], slot=None)
         for item in settings:
             key = prefix_to_add + item['name']
-            value = item['value']
             tags = {'AppService:SlotSetting': str(item['slotSetting']).lower()} if item['slotSetting'] else {}
+            value = item['value']
+
+            # Value will look like one of the following if it is a KeyVault reference:
+            # @Microsoft.KeyVault(SecretUri=https://myvault.vault.azure.net/secrets/mysecret/ec96f02080254f109c51a1f14cdb1931)
+            # @Microsoft.KeyVault(VaultName=myvault;SecretName=mysecret;SecretVersion=ec96f02080254f109c51a1f14cdb1931)
+            if value and value.lower().startswith(KeyVaultConstants.APPSVC_KEYVAULT_PREFIX.lower()):
+                try:
+                    appsvc_value_dict = dict(x.split('=') for x in value[len(KeyVaultConstants.APPSVC_KEYVAULT_PREFIX) + 1: -1].split(';'))
+                    appsvc_value_dict_lower = {k.lower(): v for k, v in appsvc_value_dict.items()}
```

Comment:
>appsvc_value_dict_lower [](start = 20, length = 23)

nit: we can reuse appsvc_value_dict

Reply:
Just like whitespacing, its not clear from appservice docs whether the keys "secreturi", "vaultname", etc, are case sensitive or not. We need to convert all keys to lowercase to be case insensitive.

---
In reply to: [364481792](https://github.com/Azure/azure-cli/pull/11773#discussion_r364481792) [](ancestors = 364481792)

Reply:
Yes, just we don't need to assign it to another variable called appsvc_value_dict_lower. 
```
appsvc_value_dicts = {k.lower(): v for k, v in appsvc_value_dict.items()}
```
Really minor comment. Up to you. 

---
In reply to: [364496405](https://github.com/Azure/azure-cli/pull/11773#discussion_r364496405) [](ancestors = 364496405,364481792)

Reply:
Makes sense. Changed variable name.

---
In reply to: [364925760](https://github.com/Azure/azure-cli/pull/11773#discussion_r364925760) [](ancestors = 364925760,364496405,364481792)